vllm/model_executor/layers/quantized_linear/base_config.py

```diff
@@ -0,0 +1,52 @@
+from typing import Any, Dict, List
+
+import torch
+
+from vllm.model_executor.layers.linear import LinearMethodBase
+
+
+class QuantizationConfig:
+    """Base class for quantization configs."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        """Name of the quantization method."""
+        raise NotImplementedError
+
+    @classmethod
+    def get_supported_act_dtypes(cls) -> List[torch.dtype]:
+        """List of supported activation dtypes."""
+        raise NotImplementedError
+
+    @classmethod
+    def get_min_capability(cls) -> int:
+        """Minimum GPU capability to support the quantization method.
+
+        E.g., 70 for Volta, 75 for Turing, 80 for Ampere.
+        This requirement is due to the custom CUDA kernels used by the
+        quantization method.
+        """
+        raise NotImplementedError
```

Comment:
Actually, we might want to define these methods as instance methods (which takes `self` as the first argument), because their outputs may depend on the configuration. For example, the supported GPUs can be different depending on the quantization bit widths.